ui/js/graphs.js

```diff
@@ -720,16 +721,44 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
   $scope.loadingTestData = false;
 
   var testInputCreated = false;
-
+  var testArray = [];
+  var series = [];
   $scope.addTestData = function () {
-    var series = _.clone($scope.selectedSeries);
-    series.projectName = $scope.selectedProject.name;
-    $modalInstance.close(series);
+      var selectedSeries = $scope.rightList
+      for (var i = 0; i < selectedSeries.length; i++) {
+          var result = testArray.filter(function (obj) {
+              return obj.name == selectedSeries[i];
+          });
+          series[i] = _.clone(result[0]); 
+          series[i].projectName = $scope.selectedProject.name;
+      }
+      $modalInstance.close(series);
   };
 
   $scope.cancel = function () {
     $modalInstance.dismiss('cancel');
   };
+  $scope.leftSelected = [];
```

Comment:
Likewise, let's rename leftSelected to something that describes better what this does, like maybe just testList.